### PR TITLE
docs: add netlify book configuration

### LINKED ISSUE #335: Create a documentation site for Secrets Store CSI Driver
**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Kubernetes version: (use `kubectl version`):

/kind documentation
/kind feature

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:
- Adds Netlify book configuration 

Most of this has been copied from https://github.com/kubernetes-sigs/cluster-api/tree/master/docs/book

Following steps from https://github.com/kubernetes/community/blob/master/github-management/subproject-site-requests.md#requesting-a-netlify-site

This adds the initial framework for the docs site.

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #335 

**Special notes for your reviewer**:

### CODE REVIEW

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Netlify build instructions
+[build]
+    command = "./docs/book/install-and-build.sh"
+    publish = "docs/book/book"
+
+# Standard Netlify redirects
+[[redirects]]
+    from = "https://master--kubernetes-sigs-secrets-store-csi-driver.netlify.com/*"
```
Comment (mrbobbytables) [inline netlify.toml:8]:
```suggestion
    from = "https://master--kubernetes-sigs-secrets-store-csi-driver.netlify.app/*"
```
Netlify no longer uses .com for their addresses - its been switched to .app

File: netlify.toml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# HTTP-to-HTTPS rules
+[[redirects]]
+    from = "http://master.secrets-store-csi-driver.sigs.k8s.io/*"
+    to = "https://master.secrets-store-csi-driver.sigs.k8s.io/:splat"
+    status = 301
+    force = true
+
+[[redirects]]
+    from = "http://master--kubernetes-sigs-secrets-store-csi-driver.netlify.com/*"
```
Comment (mrbobbytables) [inline netlify.toml:21]:
```suggestion
    from = "http://master--kubernetes-sigs-secrets-store-csi-driver.netlify.app/*"
```
Same as above, netlify no longer uses .com for hosted sites. It's now .app

**Review by mrbobbytables — CHANGES_REQUESTED** [review]
A few comments on the netlify config

File: netlify.toml (hunk 1 above)
Comment (aramase) [reply netlify.toml]: Updated!

File: netlify.toml (hunk 2 above)
Comment (aramase) [reply netlify.toml]: Updated!

**Pull request merged**